pkg/verifier/notation/revocationfactory.go

```diff
@@ -0,0 +1,32 @@
+// Copyright The Ratify Authors.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package notation
+
+import (
+	"net/http"
+
+	"github.com/notaryproject/notation-core-go/revocation"
+	corecrl "github.com/notaryproject/notation-core-go/revocation/crl"
+)
+
+type revocationFactory interface {
+	// NewFetcher returns a new fetcher instance
+	NewFetcher(*http.Client) (*corecrl.HTTPFetcher, error)
```

Comment:
Not sure if the factory is just for fetching from remote server, if it will be extended to RawBytes fetcher, the input and output should be reconsidered.

Reply:
As discussed, replaced the return with `corecrl.Fetcher` and place the http